src/layout/dom.ts

```diff
@@ -21,9 +24,24 @@ export function reset(): void {
     selectorMap = [];
     urlMap = {};
     idMap = new WeakMap();
+    regionMap = new WeakMap();
     if (Constant.DEVTOOLS_HOOK in window) { window[Constant.DEVTOOLS_HOOK] = { get, getNode, history }; }
 }
 
+export function regions(root: ParentNode): void {
+    for (let key in config.regions) {
+        if (config.regions[key] && "querySelectorAll" in root) {
+            let elements = root.querySelectorAll(config.regions[key]);
+            let length = elements.length;
+            for (let i = 0; i < length; i++) {
+                if (!(key in regionTracker)) { regionTracker[key] = 0; }
```

Comment:
>regionTracker [](start = 29, length = 13)

Can you add a comment explaining what this line with regionMap is doing? Why is a weak map?